components/discord_webhook/actions/send-message-advanced/send-message-advanced.js

```diff
@@ -1,58 +1,49 @@
-const discordWebhook = require("../../discord_webhook.app.js");
+const common = require("../send-message-common");
 
 module.exports = {
+  ...common,
   key: "discord_webhook-send-message-advanced",
   name: "Send Message (Advanced)",
   description: "Send a simple or structured message (using embeds) to a Discord channel",
-  version: "0.1.4",
+  version: "0.2.0",
   type: "action",
   props: {
-    discordWebhook,
+    ...common.props,
     message: {
       propDefinition: [
-        discordWebhook,
+        common.props.discordWebhook,
         "message",
       ],
       optional: true,
     },
-    threadID: {
-      propDefinition: [
-        discordWebhook,
-        "threadID",
-      ],
-    },
     embeds: {
       propDefinition: [
-        discordWebhook,
+        common.props.discordWebhook,
         "embeds",
       ],
     },
-    username: {
-      propDefinition: [
-        discordWebhook,
-        "username",
-      ],
-    },
-    avatarURL: {
-      propDefinition: [
-        discordWebhook,
-        "avatarURL",
-      ],
-    },
   },
   async run() {
-    const content = this.message;
     const {
+      message: content,
       avatarURL,
-      embeds,
       threadID,
       username,
+      includeSendMessageViaPipedreamFlag,
     } = this;
+    let { embeds } = this;
 
     if (!content && !embeds) {
       throw new Error("This action requires at least 1 message OR embeds object. Please enter one or the other above.");
     }
 
+    if (includeSendMessageViaPipedreamFlag) {
+      embeds = [
+        ...(embeds ?? []),
+        this.makeSentViaPipedreamEmbed(),
```

Comment:
This adds an [embedded object ](https://discord.com/developers/docs/resources/channel#embed-object), containing the "sent via Pipedream" text in its `description`, to the message's array of [`embeds`](https://birdie0.github.io/discord-webhooks-guide/discord_webhook.html).
With the embed, the Discord message looks like:
<img width="188" alt="discord-sent-via" src="https://user-images.githubusercontent.com/19861096/156486673-f5b67e1b-ec4d-40a4-b951-68f7fe711988.png">

For comparison, this message is from a previous commit, and has "sent via Pipedream" appended to the message content:
<img width="188" alt="discord-sent-via-text" src="https://user-images.githubusercontent.com/19861096/156487599-dd6817bf-4f93-4c85-827c-b8448ff66f39.png">

Reply:
Nice, thanks for putting that together! I think I like the original version better, but I'll run it by Tod tomorrow to get his take and let you know.

Reply:
Ok let's go w/ the original option of 2 line breaks (but no italics). I think you have this now as `getSentViaPipedreamText`, right? Let's go with that but just omit the surrounding `*...*`

Reply:
I changed the "Sent via pipedream" text back to the original option without italics.

A minor comment/issue with adding the "Sent via pipedream" text to the message content - the text appears above custom embeds.
For example:
<img width="425" alt="send-via-pipedream-w-embed" src="https://user-images.githubusercontent.com/19861096/156636808-85e8c8ad-ab7f-4ad9-b1cc-6aa939ea8011.png">


With this **Embeds** prop in **Send Message Advanced**
<img width="1260" alt="embeds" src="https://user-images.githubusercontent.com/19861096/156636283-1bfeb8a4-319b-4bf9-a108-2a01f49a2eed.png">

Reply:
Ah good catch --  I hadn't tested sending text **and** an embed. That isn't ideal but I think it's acceptable.